contracts/Erc20Quest.sol

```diff
@@ -29,38 +29,41 @@ contract Erc20Quest is Quest {
         rewardTokenAddress_,
         endTime_,
         startTime_,
-        totalAmount_,
+        totalParticipants_,
         allowList_,
         rewardAmountInWeiOrTokenId_,
         questId_,
         receiptContractAddress_
     ) {
         questFee = questFee_;
-        totalRedeemers = rewardAmountInWeiOrTokenId / totalAmount;
+        totalRedeemers = totalParticipants / rewardAmountInWeiOrTokenId_;
         protocolFeeRecipient = protocolFeeRecipient_;
         questFactoryContract = QuestFactory(factoryContractAddress_);
         factoryContractAddress = factoryContractAddress_;
     }
 
-    function start() public override {
-        if (IERC20(rewardToken).balanceOf(address(this)) < (totalAmount + (totalAmount * questFee / 10_000))) revert TotalAmountExceedsBalance();
-        super.start();
+    function questFeePercentage() public view returns (uint256) {
+        return questFee / 10_000;
+    }
+
+    function maxTotalRewards() public view returns (uint256) {
+        return totalParticipants * rewardAmountInWeiOrTokenId;
     }
 
-    function mintReceipt(uint amount_, string memory questId_, bytes32 hash_, bytes memory signature_) public {
```

Comment:
@jonathandiep I removed this function here, so it continues to live on the factory contract.